jujugui/static/gui/src/app/models/legacy-handlers.js

```diff
@@ -65,14 +65,24 @@ YUI.add('juju-legacy-delta-handlers', function(Y) {
       @return {undefined} Nothing.
      */
     unitInfo: function(db, action, change) {
+      // Return a list of ports represented as "NUM/PROTOCOL", e.g. "80/tcp".
+      var convertOpenPorts = function(ports) {
+        if (!ports) {
+          return [];
+        }
+        return ports.map(port => {
+          return port.Number + '/' + port.Protocol;
+        });
+      };
+
       var unitData = {
         id: change.Name,
         charmUrl: change.CharmURL,
         service: change.Application || change.Service,
         machine: change.MachineId,
         public_address: change.PublicAddress,
         private_address: change.PrivateAddress,
-        open_ports: utils.convertOpenPorts(change.Ports),
+        open_ports: convertOpenPorts(change.Ports),
```

Comment:
So this was shared code before this change. This branch basically inlines what's legacy API specific, so that the functions in utils can only focus on new API.